Scripts/OfekQuery.js

```diff
@@ -0,0 +1,323 @@
+/*Count doesnt work for one element.
+*doesnt support more than two selectors
+* check if instace func works
+* get() works*/
+function $(name) {
+	return new ofekQuery(name);
+}
+var ofekQuery = function (query) {
+    var elements = [];
+    var constructor = function () {
+        if (testHierarchy(query)) {
+            return getAllElements(query);
+        }
+        else if(isTagAndClassQuery) {
+			return getTagAndClassElements(query);
+		}
+        else {
+            console.log("false");
+            return getRootElements(query);
+        }
+
+    }
+    //Private static methods
+    var testHierarchy = function (query) {
```

Comment:
The whole thing is pretty messy, I'd tell you to think of a hella shorter solution cuz. it was not supposed to get that messy.
In addition I'm pretty sure that within this whole mess, if I tell you to change some functionality it will take you a hell huge refactor.

Remember - we want to write code as decoupled as possible, we want to write code that will be easy to change if the functionality changes even a bit.